src/enclave-agent/src/main.rs

```diff
@@ -0,0 +1,181 @@
+        }
+        std::env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", keyprovider_config);
+
+        let mut config: String = String::new();
+        let args: Vec<String> = env::args().collect();
+        // If config file specified in the args, read contents from config file
+        let config_position = args.iter().position(|a| a == "--config" || a == "-c");
+        if let Some(config_position) = config_position {
+            if let Some(config_file) = args.get(config_position + 1) {
+                config = fs::read_to_string(config_file).expect("Config file not found");
+            } else {
+                panic!("The config argument wasn't formed properly: {:?}", args);
+            }
+        }
+
+        let mut decrypt_config = None;
+        if !config.is_empty() {
+            decrypt_config = Some(config.as_str());
+        }
+
+        let source_creds = (!req.get_source_creds().is_empty()).then(|| req.get_source_creds());
+
+        let bundle_path = Path::new(CONTAINER_BASE).join(&cid);
+
+        println!("Pulling {:?}", image);
+        self.image_client
+            .lock()
+            .await
+            .pull_image(image, &bundle_path, &source_creds, &decrypt_config)
+            .await?;
+
+        Ok(image.to_owned())
+    }
+}
+
+#[async_trait]
+impl protocols::image_ttrpc::Image for ImageService {
+    async fn pull_image(
+        &self,
+        _ctx: &ttrpc::r#async::TtrpcContext,
+        req: image::PullImageRequest,
+    ) -> ttrpc::Result<image::PullImageResponse> {
+        match self.pull_image(&req).await {
+            Ok(r) => {
+                println!("Pull image {:?} successfully", r.clone());
+                let mut resp = image::PullImageResponse::new();
+                resp.image_ref = r;
+                return Ok(resp);
+            }
+            Err(e) => {
+                return Err(ttrpc_error(ttrpc::Code::INTERNAL, e.to_string()));
+            }
+        }
+    }
+}
+
+#[tokio::main(worker_threads = 1)]
+async fn main() {
+    let matches = App::new("Enclave agent")
+        .version(crate_version!())
+        .author(crate_authors!())
+        .arg(
+            Arg::with_name("listen")
+                .short("l")
+                .long("listen")
+                .value_name("sockaddr")
+                .help(&format!(
+                    "{}{}",
+                    "Specify the socket listen addr. Default is ", SOCK_ADDR
+                ))
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("decrypt config")
+                .short("c")
+                .long("config")
+                .help(&format!("The decrypt config file path"))
+                .takes_value(true)
+                .required(false),
+        )
+        .get_matches();
+
+    let mut sockaddr = SOCK_ADDR;
+    if matches.is_present("listen") {
+        sockaddr = matches.value_of("listen").unwrap();
+    }
```

Comment:
You can avoid the `mut` and `unwrap()` using something like:

```suggestion
    let sockaddr = if let Some(addr) = matches.is_present("listen") {
        addr
    } else {
        SOCK_ADDR
    };
```

Reply:
Ok, get it.